### ISSUE TITLE
cannot work with INSTALL.asciidoc's instructions

### ISSUE THREAD

**Issue by henrich** [os-autoinst/os-autoinst#580]
It says "sudo zypper in openQA" but failed.

should add
$ sudo zypper ar -f obs://devel:openQA/openSUSE_Leap_42.1 openQA
$ sudo zypper ar -f obs://devel:openQA:Leap:42.1/openSUSE_Leap_42.1 openQA-perl-modules
$ sudo zypper in openQA

then, we need git for cloning code from GitHub
$ sudo zypper in git

before configure & make, we should install make
$ sudo zypper in make

and there's some Perl modules are lucking
$ sudo zypper in perl-IPC-System-Simple
$ sudo zypper in perl-class
$ sudo zypper in Perl-exception-class

**Comment by coolo** [comment]
we should simply erase this.

**Comment by okurz** [comment]
See #582 for corrections. The perl dependencies are properly listed in the cpanfile. You would need to either call the installation commands on all dependencies or call e.g. `cpanm --installdeps .`

**Comment by henrich** [comment]
Thanks!

And it's better to note about perl modules installation since some users are not familiar with Perl and don't know about package manager.

**Comment by henrich** [comment]
Could you tell me more, please?

AS instruction, create vars.json and adjust "CASEDIR" : "/home/henrich/distri/opensuse/"

> henrich@linux-zaqb:/tmp/os-autoinst-run> ~/os-autoinst/isotovideo 
> 13:40:39.3994 2179 got sigchld
> 13:40:39.3995 2179 git hash of test distribution: 923b5afeacb2fdfae969557635693818705b07f0
> Can't locate /home/henrich/distri/opensuse//main.pm in @INC (you may need to install the ::home::henrich::distri::opensuse::::main module) (@INC contains: /tmp/os-autoinst-run/blib/arch /tmp/os-autoinst-run/blib/lib /home/henrich/os-autoinst /usr/lib/perl5/site_perl/5.18.2/x86_64-linux-thread-multi /usr/lib/perl5/site_perl/5.18.2 /usr/lib/perl5/vendor_perl/5.18.2/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.18.2 /usr/lib/perl5/5.18.2/x86_64-linux-thread-multi /usr/lib/perl5/5.18.2 /usr/lib/perl5/site_perl .) at /home/henrich/os-autoinst/isotovideo line 209.
> 13:40:39.4006 2179 awaiting death of commands process
> 13:40:39.4009 2181 signalhandler got TERM - loop 1
> [Wed Sep  7 22:40:39 2016] [info] Daemon reachable under http://*:15223/QWGRwisOLy/
> [Wed Sep  7 22:40:39 2016] [info] Listening at "http://*:15223"
> Server available at http://127.0.0.1:15223


then change "CASEDIR" : "/home/henrich/distri/opensuse/products/sle",

> henrich@linux-zaqb:/tmp/os-autoinst-run> ~/os-autoinst/isotovideo 
> 13:43:18.7807 2244 got sigchld
> 13:43:18.7808 2244 git hash of test distribution: 923b5afeacb2fdfae969557635693818705b07f0
> [Wed Sep  7 22:43:18 2016] [info] Daemon reachable under http://*:15223/QWGRwisOLy/
> [Wed Sep  7 22:43:18 2016] [info] Listening at "http://*:15223"
> Server available at http://127.0.0.1:15223
> Can't locate /home/henrich/distri/opensuse/products/sle/lib/susedistribution.pm in @INC (you may need to install the ::home::henrich::distri::opensuse::products::sle::lib::susedistribution module) (@INC contains: /home/henrich/distri/opensuse/products/sle/../../lib /tmp/os-autoinst-run/blib/arch /tmp/os-autoinst-run/blib/lib /home/henrich/os-autoinst /usr/lib/perl5/site_perl/5.18.2/x86_64-linux-thread-multi /usr/lib/perl5/site_perl/5.18.2 /usr/lib/perl5/vendor_perl/5.18.2/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.18.2 /usr/lib/perl5/5.18.2/x86_64-linux-thread-multi /usr/lib/perl5/5.18.2 /usr/lib/perl5/site_perl .) at /home/henrich/distri/opensuse/products/sle/main.pm line 103.
> Compilation failed in require at /home/henrich/os-autoinst/isotovideo line 209.
> 13:43:18.8201 2244 awaiting death of commands process
> 13:43:18.8202 2244 commands process exited: 2246

What should I set in CASEDIR? or need more settings for modules?

**Comment by okurz** [comment]
try to set PRODUCTDIR to `/home/henrich/distri/opensuse/products/opensuse/`

What are you trying to achieve in general? Try IRC in irc://chat.freenode.net/opensuse-factory for easier openQA related support. Normally `os-autoinst` is used within openQA and you don't need to mangle with `vars.json` manually. Try http://open.qa/ for a generic entry point to documentation

**Comment by henrich** [comment]
set /home/henrich/distri/opensuse/products/opensuse/ but still get errors.

> henrich@linux-zaqb:/tmp/os-autoinst-run> ~/os-autoinst/isotovideo 
> 14:52:19.5493 3581 got sigchld
> 14:52:19.5494 3581 git hash of test distribution: 923b5afeacb2fdfae969557635693818705b07f0
> Can't locate /home/henrich/distri/opensuse/products/opensuse/lib/susedistribution.pm in @INC (you may need to install the ::home::henrich::distri::opensuse::products::opensuse::lib::susedistribution module) (@INC contains: /home/henrich/distri/opensuse/products/opensuse/../../lib /tmp/os-autoinst-run/blib/arch /tmp/os-autoinst-run/blib/lib /home/henrich/os-autoinst /usr/lib/perl5/site_perl/5.18.2/x86_64-linux-thread-multi /usr/lib/perl5/site_perl/5.18.2 /usr/lib/perl5/vendor_perl/5.18.2/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.18.2 /usr/lib/perl5/5.18.2/x86_64-linux-thread-multi /usr/lib/perl5/5.18.2 /usr/lib/perl5/site_perl .) at /home/henrich/distri/opensuse/products/opensuse/main.pm line 81.
> Compilation failed in require at /home/henrich/os-autoinst/isotovideo line 209.
> 14:52:19.5586 3581 awaiting death of commands process
> 14:52:19.5591 3583 signalhandler got TERM - loop 1
> [Thu Sep  8 23:52:19 2016] [info] Daemon reachable under http://*:15223/QWGRwisOLy/
> [Thu Sep  8 23:52:19 2016] [info] Listening at "http://*:15223"
> Server available at http://127.0.0.1:15223

I want to try to introduce openqa to Debian, already put os-autoinst into archive.
https://tracker.debian.org/pkg/os-autoinst

A year ago, it works without any error but I cannot make it work well now. To clarify the problem is whether Debian specific or not,  checking install instruction.

**Comment by okurz** [comment]
On Thursday 08 September 2016 07:59:09 Hideki Yamane wrote:
> set /home/henrich/distri/opensuse/products/opensuse/ but still get errors.
> 
> > henrich@linux-zaqb:/tmp/os-autoinst-run> ~/os-autoinst/isotovideo
> > 14:52:19.5493 3581 got sigchld
> > 14:52:19.5494 3581 git hash of test distribution:
> > 923b5afeacb2fdfae969557635693818705b07f0 Can't locate
> > /home/henrich/distri/opensuse/products/opensuse/lib/susedistribution.pm

You need to set both the following for this to work:

```
CASEDIR=/home/henrich/distri/opensuse/
PRODUCTDIR=/home/henrich/distri/opensuse/products/opensuse/
```

CASEDIR is the absolute path to the test case distribution, i.e. the git 
working copy of os-autoinst-distri-opensuse.
PRODUCTDIR is the absolute path to the 'main.pm' within a test case 
distribution if not in CASEDIR.

You might also want to take a look into https://github.com/os-autoinst/os-autoinst-distri-example/ for a bare minimum test distribution which is usable.

> I want to try to introduce openqa to Debian, already put os-autoinst into
> archive. https://tracker.debian.org/pkg/os-autoinst

ok, great. Maybe the openSUSE build instructions for RPM packages help to 
understand what's necessary:
https://build.opensuse.org/project/show/devel:openQA
e.g. the os-autoinst spec file: 
https://build.opensuse.org/package/view_file/devel:openQA/os-autoinst/os-autoinst.spec?expand=1
